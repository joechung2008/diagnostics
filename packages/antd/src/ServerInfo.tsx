import { Table } from "antd";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  const columns = [
    {
      title: "Name",
      dataIndex: "name",
      key: "name",
    },
    {
      title: "Value",
      dataIndex: "value",
      key: "value",
    },
  ];

  return (
    <Table
      aria-label="Server Info"
      columns={columns}
      dataSource={items.map((item) => ({ ...item, key: item.name }))}
      pagination={false}
    />
  );
};

export default ServerInfo;
