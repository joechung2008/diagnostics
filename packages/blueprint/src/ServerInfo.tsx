import { HTMLTable } from "@blueprintjs/core";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <HTMLTable aria-label="Server Info">
      <thead>
        <tr>
          <th>Name</th>
          <th>Value</th>
        </tr>
      </thead>
      <tbody>
        {items.map((item) => (
          <tr key={item.name}>
            <td>{item.name}</td>
            <td>{item.value}</td>
          </tr>
        ))}
      </tbody>
    </HTMLTable>
  );
};

export default ServerInfo;
