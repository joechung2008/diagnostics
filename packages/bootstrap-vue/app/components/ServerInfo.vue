<template>
  <div>
    <BTable :items="rows" :fields="fields" small />
  </div>
</template>

<script setup lang="ts">
import type { ServerInfoProps } from "@/types/ServerInfoProps";
import { when } from "@/utils/extensions";

const props = defineProps<ServerInfoProps>();

const fields = [
  { key: "name", label: "Name" },
  { key: "value", label: "Value" }
];

const rows = computed(() => [
  { name: "Hostname", value: props.hostname },
  ...when(props.uptime !== undefined, {
    name: "Uptime",
    value: String(props.uptime)
  }),
  { name: "Server ID", value: props.serverId },
  { name: "Deployment ID", value: props.deploymentId },
  ...when(props.nodeVersions !== undefined, {
    name: "Node Versions",
    value: props.nodeVersions
  }),
  {
    name: "Extension Sync | Total Sync All Count",
    value: String(props.extensionSync.totalSyncAllCount)
  }
]);
</script>
