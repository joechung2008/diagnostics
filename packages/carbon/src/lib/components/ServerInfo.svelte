<script lang="ts">
	import { DataTable } from 'carbon-components-svelte';

	const { deploymentId, extensionSync, hostname, nodeVersions, serverId, uptime }: ServerInfoProps =
		$props();

	const headers = [
		{ key: 'name', value: 'Name', empty: false },
		{ key: 'value', value: 'Value', empty: false }
	];

	const rows = $derived([
		{ id: 'hostname', name: 'Hostname', value: hostname },
		{ id: 'uptime', name: 'Uptime', value: uptime },
		{ id: 'server-id', name: 'Server ID', value: serverId },
		{ id: 'deployment-id', name: 'Deployment ID', value: deploymentId },
		{ id: 'node-versions', name: 'Node Versions', value: nodeVersions },
		{
			id: 'extension-sync',
			name: 'Extension Sync | Total Sync All Count',
			value: extensionSync.totalSyncAllCount
		}
	]);
</script>

<DataTable aria-label="Server Information" {headers} {rows} />
