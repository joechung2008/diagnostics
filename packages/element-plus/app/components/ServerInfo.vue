<template>
    <el-table :data="rows" style="width: 100%" aria-label="Server Info">
        <el-table-column prop="name" label="Name" show-overflow-tooltip />
        <el-table-column prop="value" label="Value" show-overflow-tooltip />
    </el-table>
</template>

<script setup lang="ts">
import type { ServerInfoProps } from "@/types"

const props = defineProps<ServerInfoProps>()

const rows = computed(() => [
    { name: "Hostname", value: props.hostname },
    { name: "Uptime", value: props.uptime },
    { name: "Server ID", value: props.serverId },
    { name: "Deployment ID", value: props.deploymentId },
    { name: "Node Versions", value: props.nodeVersions },
    {
        name: "Extension Sync | Total Sync All Count",
        value: props.extensionSync.totalSyncAllCount
    }
])
</script>
