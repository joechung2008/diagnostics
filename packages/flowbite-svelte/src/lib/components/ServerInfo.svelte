<script lang="ts">
  import {
    Table,
    TableBody,
    TableBodyCell,
    TableBodyRow,
    TableHead,
    TableHeadCell
  } from "flowbite-svelte";

  let { deploymentId, extensionSync, hostname, nodeVersions, serverId, uptime }: ServerInfoProps =
    $props();

  const items = $derived.by(() => [
    { name: "Hostname", value: hostname },
    { name: "Uptime", value: uptime },
    { name: "Server ID", value: serverId },
    { name: "Deployment ID", value: deploymentId },
    { name: "Node Versions", value: nodeVersions },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount
    }
  ]);
</script>

<Table aria-label="Server Info">
  <TableHead>
    <TableHeadCell>Name</TableHeadCell>
    <TableHeadCell>Value</TableHeadCell>
  </TableHead>
  <TableBody>
    {#each items as item (item.name)}
      <TableBodyRow>
        <TableBodyCell>{item.name}</TableBodyCell>
        <TableBodyCell>{item.value}</TableBodyCell>
      </TableBodyRow>
    {/each}
  </TableBody>
</Table>
