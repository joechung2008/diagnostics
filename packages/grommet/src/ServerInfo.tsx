import { Table, TableBody, TableCell, TableHeader, TableRow } from "grommet";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <Table aria-label="Server Info">
      <TableHeader>
        <TableRow>
          <TableCell scope="col">Name</TableCell>
          <TableCell scope="col">Value</TableCell>
        </TableRow>
      </TableHeader>
      <TableBody>
        {items.map((item) => (
          <TableRow key={item.name}>
            <TableCell>{item.name}</TableCell>
            <TableCell>{item.value}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};

export default ServerInfo;
