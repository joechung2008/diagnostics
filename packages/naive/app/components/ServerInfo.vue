<template>
  <n-data-table :columns="columns" :data="rows" size="small" />
</template>

<script setup lang="ts">
import type { ServerInfoProps } from "@/types/ServerInfoProps"
import { when } from "~/utils/extensions"

const props = defineProps<ServerInfoProps>()

const columns = [
  { title: "Name", key: "name" },
  { title: "Value", key: "value" }
]

const rows = computed(() => [
  { name: "Hostname", value: props.hostname },
  ...when(props.uptime !== undefined, {
    name: "Uptime",
    value: String(props.uptime)
  }),
  { name: "Server ID", value: props.serverId },
  { name: "Deployment ID", value: props.deploymentId },
  ...when(props.nodeVersions !== undefined, {
    name: "Node Versions",
    value: props.nodeVersions
  }),
  {
    name: "Extension Sync | Total Sync All Count",
    value: String(props.extensionSync.totalSyncAllCount)
  }
])
</script>
