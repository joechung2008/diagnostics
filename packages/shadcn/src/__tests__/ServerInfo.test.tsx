import ServerInfo from "@/ServerInfo";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("ServerInfo", () => {
  it("matches snapshot with sample data", () => {
    const props = {
      deploymentId: "deploy-123",
      extensionSync: { totalSyncAllCount: 42 },
      hostname: "server.example.com",
      nodeVersions: "v18.19.0",
      serverId: "srv-abc",
      uptime: 987654,
    };
    const { asFragment } = render(<ServerInfo {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with empty/minimal data", () => {
    const props = {
      deploymentId: "",
      extensionSync: { totalSyncAllCount: 0 },
      hostname: "",
      nodeVersions: "",
      serverId: "",
      uptime: 0,
    };
    const { asFragment } = render(<ServerInfo {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
