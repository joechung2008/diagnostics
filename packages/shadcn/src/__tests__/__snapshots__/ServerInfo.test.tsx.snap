// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerInfo > matches snapshot with empty/minimal data 1`] = `
<DocumentFragment>
  <div
    class="relative w-full overflow-x-auto"
    data-slot="table-container"
  >
    <table
      aria-label="Server Info"
      class="w-full caption-bottom text-sm"
      data-slot="table"
    >
      <thead
        class="[&_tr]:border-b"
        data-slot="table-header"
      >
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <th
            class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-head"
          >
            Name
          </th>
          <th
            class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-head"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
        data-slot="table-body"
      >
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Hostname
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          />
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Uptime
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            0
          </td>
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Server ID
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          />
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Deployment ID
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          />
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Node Versions
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          />
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Extension Sync | Total Sync All Count
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`ServerInfo > matches snapshot with sample data 1`] = `
<DocumentFragment>
  <div
    class="relative w-full overflow-x-auto"
    data-slot="table-container"
  >
    <table
      aria-label="Server Info"
      class="w-full caption-bottom text-sm"
      data-slot="table"
    >
      <thead
        class="[&_tr]:border-b"
        data-slot="table-header"
      >
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <th
            class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-head"
          >
            Name
          </th>
          <th
            class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-head"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
        data-slot="table-body"
      >
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Hostname
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            server.example.com
          </td>
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Uptime
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            987654
          </td>
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Server ID
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            srv-abc
          </td>
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Deployment ID
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            deploy-123
          </td>
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Node Versions
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            v18.19.0
          </td>
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            Extension Sync | Total Sync All Count
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            42
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
